sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateUnsafeRowJoiner.scala

```diff
@@ -88,7 +112,20 @@ object GenerateUnsafeRowJoiner extends CodeGenerator[(StructType, StructType), U
           s"$getLong(obj2, offset2 + ${(i - bitset1Words) * 8})"
         }
       }
-      s"$putLong(buf, ${offset + i * 8}, $bits);"
+      s"$putLong(buf, ${offset + i * 8}, $bits);\n"
+    }
+
+    val functions = mutable.ArrayBuffer.empty[String]
+    val args = "java.lang.Object obj1, long offset1, java.lang.Object obj2, long offset2"
+    val copyBitsets = splitPlatformCode(copyBitset).zipWithIndex.map { case(body, index) =>
```

Comment:
why can't we use `splitExpressions` here?

Reply:
We do not have `ctx` here.

Reply:
then create one. The benefit is code reuse and the nested class optimization

Reply:
Good idea, I forgot that we can create it.